All Cortex-M platforms broken on current master

Just wanted to bulid something for the `arduino-due` on the current master, but something broke (actually it broke for all cortex-m based boards):
```
/home/hauke/dev/riot/RIOT/cpu/cortexm_common/vectors_cortexm.c: In function `hard_fault_handler':
/home/hauke/dev/riot/RIOT/cpu/cortexm_common/vectors_cortexm.c:276:9: error: `orig_sp' may be used uninitialized in this function [-Werror=maybe-uninitialized]
         __asm__ volatile (
         ^
/home/hauke/dev/riot/RIOT/cpu/cortexm_common/vectors_cortexm.c:271:15: error: `pc' may be used uninitialized in this function [-Werror=maybe-uninitialized]
         printf("In GDB:\n  set $pc=0x%" PRIx32 "\n  frame 0\n  bt\n", pc);
               ^
cc1: all warnings being treated as errors
```
What happened here?

--- a/cpu/cortexm_common/vectors_cortexm.c
+++ b/cpu/cortexm_common/vectors_cortexm.c
@@ -205,8 +205,11 @@ __attribute__((used)) void hard_fault_handler(uint32_t* sp, uint32_t corrupted, 
     uint32_t dfsr  = SCB->DFSR;
     uint32_t afsr  = SCB->AFSR;
 #endif
-    uint32_t pc;
-    uint32_t* orig_sp;
+
+    /* Initialize these variables even if they're never used uninitialized.
+     * Fixes wrong compiler warning by gcc < 6.0. */
+    uint32_t pc = 0;
+    uint32_t* orig_sp = NULL;
 
     /* Check if the ISR stack overflowed previously. Not possible to detect
      * after output may also have overflowed it. */

